### PR TITLE
[ADVAPP-621]: Introduce Report Library coming soon page under Reporting

### PR DESCRIPTION
**Pull request by payal-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-621

### Technical Description

> Introduce Report Library coming soon page under Reporting.

### CODE REVIEW

**Review by Orrison — CHANGES_REQUESTED** [review]
I apologize, I forgot that we will need permissions created and enforced for this page.

Because we do not yet know the purpose of this feature we can just still with a basic `view-any` permission.

So I would create a permission called `report-library.view-any` under the PermissionGroup "Report", which should already exist.

I have yet to update the documentation to describe how to create permissions, but you can use `database/migrations/2024_05_07_143413_seed_permissions_add_manage_s3_settings.php` as an example.

Keep in mind the migration name structure is `seed_permissions_[DESCRIPTION_OF_WHAT_IS_BEING_ADDED]`

You can follow pretty much exactly what that migration does, the main difference is that you do not need to create the Group, it should already exist.

Then once the permission is created, you can add a `canAccess` method to the Page, which you can see examples of how that is done throughout the codebase.

Let me know if you have any questions!

File: app-modules/ai/src/Enums/AiApplication.php
[the remark this answers is not in the document]
Comment (payal-canyon) [reply app-modules/ai/src/Enums/AiApplication.php:1]: Done.

File: app-modules/report/src/Filament/Pages/ReportLibrary.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+*/
+
+namespace AdvisingApp\Report\Filament\Pages;
+
+use Filament\Pages\Page;
+use AdvisingApp\Ai\Enums\AiApplication;
+
+class ReportLibrary extends Page
+{
+    public const APPLICATION = AiApplication::ReportLibrary;
```
[the remark this answers is not in the document]
Comment (payal-canyon) [reply app-modules/report/src/Filament/Pages/ReportLibrary.php]: Done.

**Review by mxm1070 — CHANGES_REQUESTED** [review]
Apologies, but we actually have a newer way to create permissions than the example @Orrison gave you. Please see this pr from Aiding: https://github.com/canyongbs/aidingapp/pull/82/files. 

You don't have to follow the exact structure, but please use the trait. Add a down method that handles deleting the permissions. I recommend updating the `CanModifyPermissions` trait like we did in Aiding because it makes deleting easier.

**Comment by payal-canyon** [comment]
> Apologies, but we actually have a newer way to create permissions than the example @Orrison gave you. Please see this pr from Aiding: https://github.com/canyongbs/aidingapp/pull/82/files.
> 
> You don't have to follow the exact structure, but please use the trait. Add a down method that handles deleting the permissions. I recommend updating the `CanModifyPermissions` trait like we did in Aiding because it makes deleting easier.

Done.

**Review by mxm1070 — APPROVED** [review]

**Review by Orrison — APPROVED** [review]

**Pull request merged**